### PR TITLE
DDF-2152 - updated AbstractMetacardActionProvider's canHandle for Att…

### PR DESCRIPTION
**Pull request by kcwire**
#### What does this PR do?
Updates the list of action providers so that the metadata & thumbnail action provider will not be displayed for metacards that do not have metadata or thumbnails respectively.
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@lcrosenbu @jaymcnallie 
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@kcwire
@millerw8
#### How should this be tested?
ingest the following metacards (`ingest -t xml no-*.txt`)
[no-metadata.txt](https://github.com/codice/ddf/files/287457/no-metadata.txt)
[no-thumbnail.txt](https://github.com/codice/ddf/files/287456/no-thumbnail.txt)
#### Any background context you want to provide?
Before this change the `Export as metadata` action would fail with an exception when a metacard didn't have metadata.  This change removes that option.
#### What are the relevant tickets?
DDF-2152
#### Checklist:
- [ ] Documentation Updated
- [x] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

…ributeMetacardTransformers

### CODE REVIEW

File: catalog/rest/catalog-rest-endpoint/src/main/java/org/codice/ddf/endpoints/rest/action/ActionProviderRegistryProxy.java [hunk 1]
```diff
@@ -42,21 +46,18 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(ActionProviderRegistryProxy.class);
 
-    private Map<ServiceReference, ServiceRegistration> actionProviderRegistry =
-            new HashMap<ServiceReference, ServiceRegistration>();
+    private static final String ATTRIBUTE_NAME = "attributeName";
 
-    private BundleContext bundleContext;
+    private Map<ServiceReference<MetacardTransformer>, ServiceRegistration> actionProviderRegistry =
+            new HashMap();
```
Comment (jrnorth) [inline catalog/rest/catalog-rest-endpoint/src/main/java/org/codice/ddf/endpoints/rest/action/ActionProviderRegistryProxy.java]: Should be `new HashMap<>()`

File: catalog/rest/catalog-rest-endpoint/src/main/java/org/codice/ddf/endpoints/rest/action/ActionProviderRegistryProxy.java [hunk 2]
```diff
@@ -42,21 +46,18 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(ActionProviderRegistryProxy.class);
 
-    private Map<ServiceReference, ServiceRegistration> actionProviderRegistry =
-            new HashMap<ServiceReference, ServiceRegistration>();
+    private static final String ATTRIBUTE_NAME = "attributeName";
```
Comment (jrnorth) [inline catalog/rest/catalog-rest-endpoint/src/main/java/org/codice/ddf/endpoints/rest/action/ActionProviderRegistryProxy.java]: Unused?

File: catalog/rest/catalog-rest-endpoint/src/main/java/org/codice/ddf/endpoints/rest/action/ActionProviderRegistryProxy.java (hunk 1 above)
Comment (kcwire) [inline catalog/rest/catalog-rest-endpoint/src/main/java/org/codice/ddf/endpoints/rest/action/ActionProviderRegistryProxy.java]: done.

File: catalog/rest/catalog-rest-endpoint/src/main/java/org/codice/ddf/endpoints/rest/action/ActionProviderRegistryProxy.java (hunk 2 above)
Comment (kcwire) [inline catalog/rest/catalog-rest-endpoint/src/main/java/org/codice/ddf/endpoints/rest/action/ActionProviderRegistryProxy.java]: removed.

File: catalog/rest/catalog-rest-endpoint/src/main/java/org/codice/ddf/endpoints/rest/action/AbstractMetacardActionProvider.java [hunk 3]
```diff
@@ -105,7 +108,14 @@ public String getId() {
 
     @Override
     public <T> boolean canHandle(T subject) {
-        return subject instanceof Metacard;
+        if (subject instanceof Metacard) {
```
Comment (lcrosenbu) [inline catalog/rest/catalog-rest-endpoint/src/main/java/org/codice/ddf/endpoints/rest/action/AbstractMetacardActionProvider.java]: Style nit: Define a singular, default return value, avoiding multiple return statements.

boolean canHandle = false;
if(someCase) {
  canHandle = true;
} 
return canHandle;

Comment (rzwiefel) [inline catalog/rest/catalog-rest-endpoint/src/main/java/org/codice/ddf/endpoints/rest/action/AbstractMetacardActionProvider.java]: I'd argue the opposite. Having a variable to keep track of while reading code makes it more difficult to understand than explicit returns. explicit > implicit, imo.

Comment (rzwiefel) [inline catalog/rest/catalog-rest-endpoint/src/main/java/org/codice/ddf/endpoints/rest/action/AbstractMetacardActionProvider.java]: I would, however, recommend flatting these which also makes it more readable.
```java
if (! (subject instanceof Metacard)) {
    return false;
}

if (StringUtils.isBlank(Attribute)) {
    return true;
}

Attribute attr = ((Metacard) subject).getAttribute(attributeName);
return (attr != null && attr.getValue() != null);
```

**Comment by kcwire** [comment]
@millerw8 I'll resolve the merge conflict - please review.

**Pull request merged**